test/cpp/ext/csm/metadata_exchange_test.cc

```diff
@@ -418,67 +421,129 @@ TEST_P(MetadataExchangeTest, VerifyCsmServiceLabels) {
             "mynamespace");
 }
 
-class MeshLabelsIterableTest : public MetadataExchangeTest {
+class MeshLabelsIterableTest : public ::testing::Test {
  protected:
-  // Returns the expected size of attributes on client/server metrics (other
-  // than `grpc.client.attempt.started` and `grpc.server.call.started`). Use of
-  // this method additionally tests `Size()` method on `MeshLabelsIterable`
-  // which is unused by the OTel C++ library at present, but might be used in
-  // the future.
-  size_t ExpectedAttributesSize(bool is_client) {
-    // Create a temporary ServiceMeshLabelsInjector and MeshLabelsIterable to
-    // get the expected size of the labels.
-    grpc::internal::ServiceMeshLabelsInjector labels_injector(
-        GetParam().GetTestResource().GetAttributes());
-    grpc::internal::MeshLabelsIterable labels_iterable(
-        labels_injector.TestOnlyLocalLabels(),
-        labels_injector.TestOnlySerializedLabels().Ref());
-    return is_client
-               ? labels_iterable.Size() + /*CSM service name and namespace*/ 2 +
-                     /*Base gRPC metric labels*/ 3
-               : labels_iterable.Size() + /*Base gRPC metric labels*/ 2;
+  MeshLabelsIterableTest()
+      : local_labels_({{"csm.workload_canonical_service", "canonical_service"},
+                       {"csm.mesh_id", "mesh"}}) {}
+
+  // Creates a serialized slice with labels for metadata exchange based on \a
+  // resource.
+  static grpc_core::Slice RemoteMetadataSliceFromResource(
+      const opentelemetry::sdk::resource::Resource& resource) {
+    return grpc::internal::ServiceMeshLabelsInjector(resource.GetAttributes())
+        .TestOnlySerializedLabels()
+        .Ref();
+  }
+
+  static std::vector<std::pair<absl::string_view, absl::string_view>>
+  LabelsFromIterable(grpc::internal::MeshLabelsIterable* iterable) {
+    std::vector<std::pair<absl::string_view, absl::string_view>> labels;
+    while (true) {
+      auto label = iterable->Next();
+      if (!label.has_value()) break;
+      labels.push_back(*std::move(label));
+    }
+    EXPECT_EQ(labels.size(), iterable->Size());
+    return labels;
+  }
+
+  static std::string PrettyPrintLabels(
+      const std::vector<std::pair<absl::string_view, absl::string_view>>&
+          labels) {
+    std::string result;
```

Comment:
This doesn't appear to be needed.

Reply:
I'm surprised that the compiler did not complain